### DISCUSSION TITLE
Setting asset metadata through graphQL

### DISCUSSION THREAD

**Discussion by neha04** [pimcore/pimcore#10787] [Q&A / Support]
Hi all,

How can we set asset metadata through graph ql or pimcore default rest apis?

**Comment by dvesh3** [comment]
You need to provide update permission for assets in Data Hub GraphQL API endpoint:
![image](https://user-images.githubusercontent.com/5137917/140922735-e9fc78e5-cfa6-41e7-81e1-7359cf364be1.png)

and use mutation query:
```
mutation {
  updateAsset(
    id: 67
    input: { 
      metadata: [
        { name: "txt", type: "input", data: "this is test" },
        { name: "obj", type: "object", data: "1" }
      ] 
    }
  ) {
    success
    message
  }
}

```

Doc link: https://pimcore.com/docs/data-hub/current/GraphQL/Mutation/Asset_Mutations.html#page_Update-Asset

**Reply by neha04** [reply]
Thanks @dvesh3 .It Worked .Maybe we can include this in documentation as well .The example in the documentation updates only asset data.
Also is there any way to change embedded metadata also?

Whats the query to get the embedded metadata from the asset? The example in the documentaion shows how to get the custom metadata.

**Reply by dvesh3** [reply]
you can't change embedded metadata in Pimcore, so it is not possible to change through data-hub as well.

For documentation update, PRs are always welcome 😊

**Reply by neha04** [reply]
@dvesh3 what about the custom settings. We can add more custom settings in pimcore as per the documentation. I was trying to get them or add them but didnt get the method for customsettings in getAsset
